prov/shm/src/smr_init.c

```diff
@@ -63,28 +63,35 @@ static void smr_resolve_addr(const char *node, const char *service,
 	(*addr)[*addrlen - 1]  = '\0';
 }
 
-static int smr_get_ptrace_scope(void)
+static int smr_check_cma_capability(void)
 {
-	FILE *file;
-	int scope, ret;
-
-	scope = 0;
-	file = fopen("/proc/sys/kernel/yama/ptrace_scope", "r");
-	if (file) {
-		ret = fscanf(file, "%d", &scope);
-		if (ret != 1) {
+	pid_t pid;
+	int ret = 0;
+	int flag = 0;
+
+	pid = fork();
+	if (pid == 0) {
+		// child process tries to execute CMA write and exits
+		int cflag = 1;
+		struct iovec local;
+		struct iovec remote;
+
+		local.iov_base = &cflag;
+		local.iov_len = sizeof(int);
```

Comment:
Best to use `sizeof(cflag)` here and `sizeof(flag)` here

Reply:
Will fix in new revison.